src/pages/library/DataExplorer.js

```diff
@@ -1,30 +1,107 @@
-import { Fragment } from 'react'
-import { div, h } from 'react-hyperscript-helpers'
+import _ from 'lodash/fp'
+import { Component, Fragment } from 'react'
+import { div, h, span } from 'react-hyperscript-helpers'
 import IframeResizer from 'react-iframe-resizer-super'
-import TopBar from 'src/components/TopBar'
+import { ajaxCaller } from 'src/libs/ajax'
+import { authStore, contactUsActive } from 'src/libs/state'
+import { centeredSpinner } from 'src/components/icons'
+import { commonPaths } from 'src/components/breadcrumbs'
+import { link } from 'src/components/common'
 import * as Nav from 'src/libs/nav'
 import * as Style from 'src/libs/style'
+import TopBar from 'src/components/TopBar'
+import * as Utils from 'src/libs/utils'
+
+// name must be name from Data Explorer dataset.json
+// authDomain must be authorization_domain from Data Explorer dataset.json
+const datasets = [
+  {
+    name: '1000 Genomes',
+    origin: 'https://test-data-explorer.appspot.com'
+  },
+  {
+    name: 'AMP PD - 2019_v1beta_0220',
+    origin: 'https://amp-pd-data-explorer.appspot.com',
+    authDomain: 'amp-pd-researchers'
+  },
+  {
+    name: 'Baseline Health Study',
+    origin: 'https://baseline-baseline-explorer.appspot.com',
+    authDomain: 'baseline-researchers'
+  },
+  {
+    name: 'Nurses\' Health Study',
+    origin: 'https://nhs-explorer.appspot.com',
+    authDomain: 'nhs_saturn_users'
+  },
+  {
+    name: 'UK Biobank',
+    origin: 'https://biobank-explorer.appspot.com',
+    authDomain: 'Kathiresan_UKBB'
+  }
+]
 
+const DataExplorer = _.flow(
+  ajaxCaller,
+  Utils.connectAtom(authStore, 'authState')
+)(class DataExplorer extends Component {
+  constructor(props) {
+    super(props)
+    const { dataset } = props
+    this.state = {
+      completedDeOauth: undefined,
+      groups: undefined
+    }
+    this.origin = datasets.filter(d => d.name === dataset)[0].origin
+    this.authDomain = datasets.filter(d => d.name === dataset)[0].authDomain
+  }
 
-const datasetToUrl = {
-  // key must be dataset name from Data Explorer dataset.json
-  '1000 Genomes': 'https://test-data-explorer.appspot.com/?embed'
-}
+  async componentDidMount() {
+    const { ajax: { Groups } } = this.props
+    const groups = _.map(g => g.groupName, await Groups.list())
+    this.setState({ groups: groups })
 
+    try {
+      // This fetch will succeed iff user has used this Data Explorer from this
+      // browser.
+      // Need to include credentials to include IAP login cookie, if it exists.
+      await fetch(this.origin + '/favicon.ico', { credentials: 'include' })
+      this.setState({ completedDeOauth: true })
+    } catch (e) {
+      // fetch will fail if:
+      // - User has not completed oauth for this Data Explorer
+      // - User has completed oauth but has not used DE from this browser
+      // - User has used DE from this browser but IAP login cookie has expired
+      this.setState({ completedDeOauth: false })
+    }
```

Comment:
nit: This check could be done in parallel with the groups load, by splitting it out into a separate async function.